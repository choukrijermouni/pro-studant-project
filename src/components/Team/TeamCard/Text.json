{
  "without": "Sans",
  "learner": "Apprenants",
  "manager": "Managers dâ€™Ã©quipe"
}