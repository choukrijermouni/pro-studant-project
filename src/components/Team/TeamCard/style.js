import { blue } from '@pro_boa/ui'
export default {
  row: {
    margin: 16,
    display: 'flex',
    justifyContent: 'flex-start',
    minHeight: 50,
    overflow: 'hidden',
    alignItems: 'center',
    padding: 12
  },
  checkBox: {
    width: '0px !important'
  },
  cardContainer: {
    cursor: 'pointer'
  },
  card: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    width: '11vw'
  },
  ManagersCard: {
    display: 'flex',
    alignItems: 'center',
    marginLeft: 24,
    width: ({ scale }) => scale > 1 ? '21vw' : '25vw'
  },
  icon: {
    fontSize: 16,
    color: blue[1]
  },
  teamTag: {
    width: 43,
    height: 43,
    borderRadius: '50%',
    marginLeft: 16
  },
  fontStyle: {
    width: 210
  },
  teamNameClass: {
    whiteSpace: 'nowrap',
    overflow: 'hidden',
    textOverflow: 'ellipsis'
  },
  nameContainer: {
    width: '15vw',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-start'
  },
  iconSkeleton: {
    display: 'flex',
    width: 30
  },
  avatarspaceSkeleton: {
    padding: '20px 2px',
    height: 68
  },
  flex: {
    display: 'flex',
    alignItems: 'center'
  }
}
