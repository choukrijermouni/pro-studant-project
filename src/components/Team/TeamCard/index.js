import React from 'react'
import { createUseStyles } from 'react-jss'
import style from './style'
import { Paper, TypographyElement, neutral, Icon, Icons, Skeleton, Avatar, AvatarSize } from '@pro_boa/ui'
import Text from './Text.json'
import { teamDev } from 'assets'

const defaultHandleClick = () => {}

export default ({ teamIcon, name, membersCount, managers, loading, handleClick = defaultHandleClick }) => {
  const scale = window.devicePixelRatio
  const useStyle = createUseStyles(style)
  const {
    row,
    card,
    icon,
    teamTag,
    fontStyle,
    cardContainer,
    teamNameClass,
    ManagersCard,
    nameContainer,
    avatarspaceSkeleton,
    flex,
    avatarContainer,
    iconSkeleton
  } = useStyle({ scale })
  const formattedManagers = !loading && managers.map(manager => `${manager.FirstName} ${manager.LastName}`)
  return (
    loading
      ? (
        <Paper className={row}>
          <div className={flex} onClick={handleClick}>
            <div className={nameContainer}>
              <div className={avatarContainer}>
                <Avatar
                  loading
                  size={AvatarSize.size_1}
                />
              </div>
              <span className={avatarspaceSkeleton} />
              <Skeleton lines={1} height={18} width={150} margin='20px 50px 20px 2px' />
            </div>
            <div title={Text.creation} className={card}>
              <div className={iconSkeleton}>
                <Skeleton lines={1} height={18} width={18} />
              </div>
              <Skeleton lines={1} height={18} width={75} margin='20px 2px' />
            </div>
            <div title={Text.team} className={card}>
              <div className={iconSkeleton}>
                <Skeleton lines={1} height={18} width={18} />
              </div>
              <Skeleton lines={1} height={18} width={81} margin='20px 2px' />
            </div>
          </div>
        </Paper>
        )
      : (
        <div className={cardContainer} onClick={handleClick} data-test='team-card'>
          <Paper className={row}>
            <div className={nameContainer}>
              <img src={teamIcon || teamDev} alt='' className={teamTag} />
              <div className={fontStyle}>
                <TypographyElement
                  component='h3'
                  variant='heading3'
                  align='left'
                  fontSize={16}
                  spacing='20px'
                  color={neutral[6]}
                  className={teamNameClass}
                >
                  {name}
                </TypographyElement>
              </div>
            </div>
            <div title={Text.learner} className={card}>
              <Icon iconName={Icons.users} style={icon} />
              <TypographyElement
                component='h4'
                variant='caption1'
                align='left'
                spacing='10px 16px 6px 16px'
                color={neutral[4]}
              >
                {membersCount < 10 ? '0' + membersCount : membersCount}
              </TypographyElement>
            </div>
            <div title={Text.manager} className={ManagersCard}>
              <Icon iconName={Icons.admin} style={icon} />
              <TypographyElement
                component='h4'
                variant='caption1'
                align='left'
                spacing='10px 16px 6px 16px'
                color={neutral[4]}
              >
                {managers.length
                  ? formattedManagers.slice(0, 3).join(' - ')
                  : Text.without}
                {formattedManagers.length > 3 && ' . . . '}
              </TypographyElement>
            </div>
          </Paper>
        </div>
        )
  )
}
