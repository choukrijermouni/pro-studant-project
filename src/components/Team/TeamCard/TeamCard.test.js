import { shallow, configure } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import renderer from 'react-test-renderer'
import TeamCard from '.'
configure({ adapter: new Adapter() })

describe('>>> TeamCard --- Shallow Render ManagerCard component', () => {
  let wrapper
  let tree

  beforeEach(() => {
    wrapper = shallow(<TeamCard />)
    tree = renderer.create(
      <TeamCard />
    ).toJSON()
  })

  it('+++ render correctly and match snapshot', () => {
    expect(tree).toMatchSnapshot()
  })

  it('+++ render the component', () => {
    expect(wrapper).toBeTruthy()
  })
})
