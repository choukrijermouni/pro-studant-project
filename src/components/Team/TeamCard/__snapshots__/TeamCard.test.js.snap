// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> TeamCard --- Shallow Render ManagerCard component +++ render correctly and match snapshot 1`] = `
<div
  className="row-0-2-6 paper-0-2-11"
>
  <div
    className="inputGroup-0-2-12"
  >
    <input
      className="input-0-2-14"
      disabled={false}
      id="desktop1"
      type="checkbox"
    />
    <label
      className="checkbox-0-2-13 checkbox-d0-0-2-16"
    />
    <label
      className="label-0-2-15 checkBox-0-2-7"
      htmlFor="desktop1"
    >
      
    </label>
  </div>
  <img
    alt="teamTag"
    className="teamTag-0-2-10"
  />
  <p
    className="text-0-2-17 text-d0-0-2-18"
  />
  <div
    className="card-0-2-8"
  >
    <span
      className="icon-users-alph-iconfont icon-0-2-9"
    />
    <p
      className="text-0-2-19 text-d0-0-2-20"
    />
  </div>
  <div
    className="card-0-2-8"
  >
    <span
      className="icon-admins-alph-iconfont icon-0-2-9"
    />
    <p
      className="text-0-2-21 text-d0-0-2-22"
    />
  </div>
</div>
`;
